### DISCUSSION TITLE
How to warm start with expo?

### DISCUSSION THREAD

**Discussion by giautm** [expo/expo#14295] [Archived: Questions]
I trying to e2e testing with Detox. Using the example from https://github.com/fschoenfeldt/clean-expo-detox-testing

On the first time start, the bundle process takes too long and the test case failed. How I can start my app in warm mode?

<img width="485" alt="Screen Shot 2021-09-05 at 12 54 53" src="https://user-images.githubusercontent.com/12751435/132117035-167d6949-5e3f-4652-bcdd-1fd64c43e894.png">

**Comment by EvanBacon** [comment]
Interesting question. Due to the lack of metro documentation I don't know of a built-in method for this, but you could try building a script to fetch the bundle from node, which would trigger a build. This can be done by fetching `/` for the manifest.json, then parsing the JSON for `bundleUrl`, which you can fetch to trigger a build. 

`bundleUrl` usually looks like: `"bundleUrl": "http://192.168.1.64:19000/index.bundle?platform=ios&dev=true&hot=false&minify=false",` where `index.bundle` is the main file relative to the project root, but with `.bundle` instead of `.js`, `.ts`, `.tsx`, etc. The query params will need to match what your native code fetches, you'll want to change the platform parameter to the platform you're attempting to warm.

If you know of a mechanism to trigger the Metro dev server programmatically, I'd be happy to investigate adding it to expo/dev-server package in expo/expo-cli.
